### ISSUE TITLE
When will GearmanWorker::work be false?

### ISSUE THREAD

**Issue by ifohancroft** [php/pecl-networking-gearman#13]
I have looked everywhere (even tried reading the C source but that didn't help) but I can't find the answer to that.

When will GearmanWorker::work be false? What can/will cause it to "fail" basically?
The PHP manual over at php.net says: 

```
Waits for a job to be assigned and then calls the appropriate callback function. Issues an E_WARNING with the last Gearman error if the return code is not one of GEARMAN_SUCCESS, GEARMAN_IO_WAIT, or GEARMAN_WORK_FAIL.
```

If the return code of what is not one of GEARMAN_SUCCESS, GEARMAN_IO_WAIT or GEARMAN_WORK_FAIL?

Is there a way to see the Gearman Worker class as a PHP code? I.e. as if it was a class written in PHP?

**Comment by ifohancroft** [comment]
> Here is the source code for this method:
> 
> https://github.com/php/pecl-networking-gearman/blob/7da13e4babc17067b2b45d6b37041c3c8ed91637/php_gearman_worker.c#L627
> 
> It will return `false` if return code is not `GEARMAN_SUCCESS`.

Thank you! The return code of what though?

**Comment by esabol** [comment]
> The return code of what though?

The C function `gearman_worker_work` in libgearman.

**Comment by ifohancroft** [comment]
> > The return code of what though?
> 
> The C function `gearman_worker_work` in libgearman.

No, I get that. I mean like what needs to happen on a higher level in-order for $worker->work() to return false. On a low level, I know the C function needs to return false. But I'm asking on a Gearman level. Like, does the server need to throw an error, or does the function that handles the job needs to fail, etc
